CONTRIBUTING.md

```diff
@@ -23,3 +23,34 @@ use GitHub pull requests for this purpose. Consult [GitHub Help] for more
 information on using pull requests.
 
 [GitHub Help]: https://help.github.com/articles/about-pull-requests/
+
+## Building Dagger
+
+Dagger is built with [`bazel`](https://bazel.build).
+
+### Building Dagger from the command line
+
+* Go to the [Bazel installation page](https://docs.bazel.build/versions/master/install.html) and install Bazel.
+  * Learn more about Bazel targets [here](https://docs.bazel.build/versions/master/build-ref.html).
+* Clone the Dagger repository: `git clone git@github.com:google/dagger.git dagger`
+* Build the Dagger project with `bazel build`
+  * If you get the following error: `ERROR: missing input file '@androidsdk//:build-tools/26.0.2/aapt'`, install the missing build tools version with sdkmanager.
+* You can install the Dagger libraries in your **local maven repository** by running the `./util/install-local-snapshot.sh` script.
+  * It will build the libraries and install them with a `LOCAL-SNAPSHOT` version.
+* You can **run all tests** with `bazel test //...` and a single test with the path and class name, e.g. `bazel test ///javatests/dagger/internal/codegen:KeyFactoryTest`.
+
+### Importing the Dagger project in Android Studio
+
+* In *Android Studio*, go to `Preferences > Plugins`.
+  * search for `bazel` and install the plugin.
+  * If no result shows up, click on `Search in repositories`, search for `bazel` and install the plugin.
+  * The Bazel plugin is available for Android Studio 3.2 or greater.
+* Select `Import Bazel Project`.
+* Input the path to the Dagger project under `workspace`, click `Next`.
+* Select `Generate from BUILD file`, type `BUILD` in the `Build file` input, click `Next`.
+* In the `Project View` form, uncomment one of the `android_sdk_platform` lines. Pick one that you have installed, then click `Finish`.
+  * If you get an error on Bazel sync, `Cannot run program "bazel"`, then:
```

Comment:
Should probably add a note for what to do to re-sync after fixing the setting. AKA clicking this super obvious volcano? icon

![image](https://user-images.githubusercontent.com/1361086/50554209-58102d80-0c6b-11e9-8886-7cbe545f699e.png)

Reply:
This unfortunately yields the earlier `incompatible_package_name_is_a_function` issue again. Went ahead and added that .bzlrc line from above.

This yields a different error, where the sync partially finishes

```console
INFO: Invocation ID: 6bdfd484-7fe2-4e06-92dd-4a42ec46958e
INFO: Loading complete.  Analyzing...
ERROR: /private/var/tmp/_bazel_zsweers/1825c3febee87f83f0190e5ba08c9307/external/local_config_cc/BUILD:36:1: in @intellij_aspect//:intellij_info_bundled.bzl%intellij_info_aspect aspect on cc_toolchain_suite rule @local_config_cc//:toolchain: 
Traceback (most recent call last):
	File "/private/var/tmp/_bazel_zsweers/1825c3febee87f83f0190e5ba08c9307/external/local_config_cc/BUILD", line 36
		@intellij_aspect//:intellij_info_bundled.bzl%intellij_info_aspect(...)
	File "/private/var/tmp/_bazel_zsweers/1825c3febee87f83f0190e5ba08c9307/external/intellij_aspect/intellij_info_bundled.bzl", line 38, in _aspect_impl
		intellij_info_aspect_impl(target, ctx, semantics)
	File "/private/var/tmp/_bazel_zsweers/1825c3febee87f83f0190e5ba08c9307/external/intellij_aspect/intellij_info_impl.bzl", line 788, in intellij_info_aspect_impl
		collect_c_toolchain_info(target, ctx, semantics, ide_info, <2 more arguments>)
	File "/private/var/tmp/_bazel_zsweers/1825c3febee87f83f0190e5ba08c9307/external/intellij_aspect/intellij_info_impl.bzl", line 364, in collect_c_toolchain_info
		cc_common.create_compile_variables(feature_configuration = feature_..., <2 more arguments>)
Passing depset into user flags is deprecated (see --incompatible_disable_depset_in_cc_user_flags), use list instead.
INFO: Found 497 targets...
INFO: Building...
ERROR: command succeeded, but not all targets were analyzed.
INFO: Elapsed time: 22.065s, Critical Path: 21.23s
INFO: 103 processes: 95 darwin-sandbox, 8 worker.
```

So now adding this line also to `.bazelrc`

```
build --incompatible_disable_depset_in_cc_user_flags=false
```

And now the build works, with the running diff being

```patch
diff --git a/.bazelrc b/.bazelrc
index ecd54a58..5b4db51d 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -6,3 +6,7 @@ build --host_javacopt=-g
 # TODO(ronshapiro): explore how much work it would be to reenable this
 build --javacopt="-Xep:BetaApi:OFF"
 build --host_javacopt="-Xep:BetaApi:OFF"
+
+# Legacy flags
+build --incompatible_package_name_is_a_function=false
+build --incompatible_disable_depset_in_cc_user_flags=false
diff --git a/javatests/dagger/android/support/functional/BUILD b/javatests/dagger/android/support/functional/BUILD
index ffb93618..a67e7027 100644
--- a/javatests/dagger/android/support/functional/BUILD
+++ b/javatests/dagger/android/support/functional/BUILD
@@ -33,6 +33,7 @@ android_library(
         "@androidsdk//com.android.support:appcompat-v7-25.0.0",
         "@androidsdk//com.android.support:support-fragment-25.0.0",
         "@google_bazel_common//third_party/java/guava",
+        "@javax_annotation_jsr250_api",
     ],
 )
 

```